ssh/tailssh/incubator_darwin.go

```diff
@@ -19,3 +29,15 @@ func setGroups(groupIDs []int) error {
 	// this to work for more things than it previously did.
 	return syscall.Setgroups(groupIDs[:16])
 }
+
+// maybeRunCommandWithLoginDarwin is the darwin implementation of maybeRunCommandWithLogin
+func maybeRunCommandWithLoginDarwin(logf logger.Logf, ia incubatorArgs) (bool, error) {
+	if ia.isSFTP || ia.isShell || ia.cmdName == "" || ia.loginCmdPath == "" {
+		return false, nil
+	}
+
+	// if we run it this way on Mac, this seems to solve various user ID issues
+	return true, unix.Exec(ia.loginCmdPath, append(
```

Comment:
unix.Exec return value, eh? So what's the point of the bool result parameter? If it returns true, it's always an error?

Reply:
If it doesn't handle it (ie if not on Mac), then it returns false. I put that comment where the variable if is defined in `incubator.go`. You're technically right, since `unix.Exec()` is special, and I guess won't will return `nil`, I guess we could special case the method signature to mean: "return nil if unable to handle this"?

Felt simpler to me to return 2 values. Can change if you feel strongly.

I used `unix.Exec()` rather than `exec.Command()` as that matched the other invocation of `/usr/bin/login` from `incubator.go`.  It also means that the process should get killed if the connection drops.